src/drivers/imx/esai.c

```diff
@@ -197,17 +197,23 @@ static inline int esai_set_config(struct dai *dai,
 
 	xccr |= ESAI_xCCR_xHCKD; /* Set the HCKT pin as an output */
 
-	dai_update_bits(dai, REG_ESAI_TCCR, mask, xccr); /* rx */
-	dai_update_bits(dai, REG_ESAI_RCCR, mask, xccr); /* tx */
+	dai_update_bits(dai, REG_ESAI_TCCR, mask, xccr);
+	/* We must make our rx clock slave because codec is master
+	 * TODO make this come from topology
+	 */
+	xccr &= ~(ESAI_xCCR_xCKD | ESAI_xCCR_xFSD);
+	dai_update_bits(dai, REG_ESAI_RCCR, mask, xccr);
 
 	mask = ESAI_xCR_xFSL | ESAI_xCR_xFSR | ESAI_xCR_xWA |
 		ESAI_xCR_xMOD_MASK | ESAI_xCR_xSWS_MASK | ESAI_xCR_PADC |
 		ESAI_xCR_xPR;
 	/* Personal reset, suspend the actual TX/RX for now */
 	xcr |= ESAI_xCR_xPR;
 
-	dai_update_bits(dai, REG_ESAI_TCR, mask, xcr); /* rx */
-	dai_update_bits(dai, REG_ESAI_RCR, mask, xcr); /* tx */
+	dai_update_bits(dai, REG_ESAI_TCR, mask, xcr);
+	/* rx doesn't have any PADC, unset it */
+	xcr &= ~ESAI_xCR_PADC;
+	dai_update_bits(dai, REG_ESAI_RCR, mask, xcr);
 
```

Comment:
Is this necessary? Or its just a cosmetic cleanup?

Reply:
The bit that would be at the same position in RCR as PADC is in TCR is reserved and probably ignored. But I still believe reserved bits should be left alone.

Perhaps a better variant is to remove PADC from the mask itself instead?

Reply:
@paulstelian97 sure, but please make it as a separate commit. Also, this is no longer an RFC so just update the PR description to reflect this.

Reply:
Alright, doing that.